packages/react-ui-components/src/components/SchemaForm/Widgets/SelectWidget/SelectWidget.tsx

```diff
@@ -1,47 +1,72 @@
-import React, { useMemo, useState } from 'react';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
 import { WidgetProps } from '@rjsf/core';
 import { KvMultiSelectDropdown, KvSingleSelectDropdown } from '../../../stencil-generated';
 import { EValidationState } from '@kelvininc/ui-components';
-import { buildSelectedOptions, getSelectedOptions, processValue } from './utils';
+import { buildDropdownOptions, buildSelectedOptions, getSelectedOptions, processValue, searchDropdownOptions } from './utils';
 
-const SelectWidget = ({ schema, id, options, label, required, disabled, readonly, value, multiple, onChange, placeholder, rawErrors = [] }: WidgetProps) => {
+const SelectWidget = ({ schema, id, options, label, required, disabled, readonly, value, multiple, onChange, placeholder, rawErrors = [], uiSchema = {} }: WidgetProps) => {
 	const { enumOptions, enumDisabled } = options;
+	const { displayValue, searchable, selectionClearable } = uiSchema;
+
+	const [searchTerm, setSearchTerm] = useState<string | null>(null);
+
+	const defaultDropdownOptions = useMemo(() => buildDropdownOptions(enumOptions, enumDisabled), [enumOptions, enumDisabled]);
+	const searchedOptions = useMemo(() => {
```

Comment:
`searchedOptions` should have a better naming since is used to display all/searched options

Reply:
perhaps `filteredOptions`?

Reply:
Will change it 👍